cloud-aws-cloudformation/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsStackValidator.java

```diff
@@ -24,15 +44,70 @@
 
     @Override
     public void validate(AuthenticatedContext ac, CloudStack cloudStack) {
+        validateStackNameAvailability(ac);
+        validateInstanceStorageOnInstanceTypes(ac, cloudStack);
+    }
+
+    private void validateStackNameAvailability(AuthenticatedContext ac) {
         AwsCredentialView credentialView = new AwsCredentialView(ac.getCloudCredential());
         String regionName = ac.getCloudContext().getLocation().getRegion().value();
         AmazonCloudFormationClient cfClient = awsClient.createCloudFormationClient(credentialView, regionName);
         String cFStackName = cfStackUtil.getCfStackName(ac);
         try {
+            LOGGER.debug("Checking stack name availability. [{}]", cFStackName);
             cfClient.describeStacks(new DescribeStacksRequest().withStackName(cFStackName));
             throw new CloudConnectorException(String.format("Stack is already exists with the given name: %s", cFStackName));
-        } catch (AmazonServiceException ignored) {
+        } catch (AmazonServiceException e) {
+            if (e.getMessage().toLowerCase().contains("not found")) {
+                LOGGER.info("Stack name is available, CF stack not found by name {}", cFStackName);
+            } else {
+                LOGGER.warn("Exception while checking stack name availability.", e);
+            }
+        }
+    }
 
+    private void validateInstanceStorageOnInstanceTypes(AuthenticatedContext ac, CloudStack cloudStack) {
+        LOGGER.debug("Check instance storage availability on instance types");
+        List<String> instanceTypes = cloudStack.getGroups().stream()
+                .map(Group::getInstances)
+                .flatMap(Collection::stream)
+                .filter(toEphemeralStorageTemplates())
+                .map(CloudInstance::getTemplate)
+                .map(InstanceTemplate::getFlavor)
+                .collect(Collectors.toList());
+        if (CollectionUtils.isEmpty(instanceTypes)) {
+            LOGGER.debug("No instance types to check for instance storage.");
+            return;
```

Comment:
I would rather see the return removed and put the rest in an else branch. This way it would be obvious by looking at the code that there is 2 exit point